### PR TITLE
fwupd: 1.2.1 -> 1.2.3

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

The fwupd NixOS test passes :).

Also this disables the test plugin by default
for a slightly improved default experience.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 1]
```diff
@@ -19,10 +19,10 @@ in stdenv.mkDerivation {
   name = "fwupd-${version}";
   src = fetchurl {
     url = "https://people.freedesktop.org/~hughsient/releases/fwupd-${version}.tar.xz";
-    sha256 = "126b3lsh4gkyajsqm2c8l6wqr4dd7m26krz2527khmlps0lxdhg1";
+    sha256 = "11qpgincndahq96rbm2kgcy9kw5n9cmbbilsrqcqcyk7mvv464sl";
   };
 
-  outputs = [ "out" "lib" "dev" "devdoc" "man" "installedTests" ];
+  outputs = [ "out" "dev" "devdoc" "man" "installedTests" ];
```
Comment (jtojnar) [inline pkgs/os-specific/linux/firmware/fwupd/default.nix]: Weird, I am getting the circular dependency but cannot `grep` the `lib` references in the `out`

**Comment by jtojnar** [comment]
`MetadataURI` key in `fwupd/remotes.d/fwupd-tests.conf` points to wrong path (`out` output instead of `installedTests`) but fixing it will probably cause a cycle.

**Comment by dtzWill** [comment]
I think they from the wrapped binaries,
so maybe your grep invocation wasn't checking 'hidden' files?

**Comment by jtojnar** [comment]
Is seems like the issue is that the newly added plug-ins are in `lib` output instead of `out`.

```ShellSession
$ git checkout channels/nixos-unstable-small
$ du --human-readable $(nix-build -A fwupd.all) --summarize 
2,0M	/nix/store/d92li0fnkyxdyr98xccwxiw2pm2l4h8p-fwupd-1.2.1
3,7M	/nix/store/cyjma8cf68lg1rcdli55w61rlhw8lnky-fwupd-1.2.1-lib
272K	/nix/store/dh33r3xnykrhjh0nvilfaxwf6qsidslq-fwupd-1.2.1-dev
976K	/nix/store/502c6qikkfq08sy71g86bac3wqbc1kwf-fwupd-1.2.1-devdoc
24K	/nix/store/rvvrly90n0c0gqw2dqmc2w9jisdl0va4-fwupd-1.2.1-man
52K	/nix/store/0fvw17z4w3rv0kqxcpbmi43rm1vcxs81-fwupd-1.2.1-installedTests
$ nix path-info -S /nix/store/d92li0fnkyxdyr98xccwxiw2pm2l4h8p-fwupd-1.2.1 /nix/store/cyjma8cf68lg1rcdli55w61rlhw8lnky-fwupd-1.2.1-lib
/nix/store/cyjma8cf68lg1rcdli55w61rlhw8lnky-fwupd-1.2.1-lib	  232096728
/nix/store/d92li0fnkyxdyr98xccwxiw2pm2l4h8p-fwupd-1.2.1    	  570954824
$ git checkout dtzWill/update/fwupd-1.2.3
$ du --human-readable $(nix-build -A fwupd.all) --summarize 
11M	/nix/store/79dn0g3wnnghk4bx6b75vxdhmmfkamg0-fwupd-1.2.3
272K	/nix/store/9aphq7lr2k6rnim1i843n0hgli2if01y-fwupd-1.2.3-dev
992K	/nix/store/s23vnalm4gp0bgh1h6hbnjpcankiv7r8-fwupd-1.2.3-devdoc
24K	/nix/store/ahppw5jh7nbib2yhx76cxbaycf0rg1bm-fwupd-1.2.3-man
52K	/nix/store/4z7zwmnnl8mvk7w8x52qcim9ncg8lncr-fwupd-1.2.3-installedTests
$ nix path-info -S /nix/store/79dn0g3wnnghk4bx6b75vxdhmmfkamg0-fwupd-1.2.3
/nix/store/79dn0g3wnnghk4bx6b75vxdhmmfkamg0-fwupd-1.2.3	  575884792
```

**Comment by dtzWill** [comment]
Hmm, well that's easy enough to fix (pushed commit doing so).

What's the goal with the separate "lib" output here, anyway?

I'm not sure what uses fwupd (gnome-software?),
would this maybe make libfwupd users have difficulties
using the plugins?

-----

(mini-rant, don't mind me...)

BTW **gah** at another piece of software with a dynamic plugin system
that is, in practice, entirely resolved at compile-time...
and the rest of the use cases are all install/deploy-time.

So no reason to have search directories of shared libraries
that are totally-going-to-work-TM,
concern yourself with how plugins are 'discovered';
similarly no need for blacklist configuration files...
just include the ones you need?
(best way to blacklist code is to never have it on your computer in the
first place :P)

O:)

**Comment by jtojnar** [comment]
Yup, gnome-software is what I was concerned about.

Plug-ins should work fine, as the `plugin_dir` meson variable is used in for locating the plug-ins at runtime as well.

I fixed the location of the `fwupd-tests.conf` so hopefully this is ready now.

**Pull request merged**